### ISSUE TITLE
Self hosted + SSL + Android App

### ISSUE THREAD

**Issue by GastroGeek** [standardnotes/forum#454]
I have managed to set up self-hosted server and the macOS/android apps... **without** SSL... as soon as I add in an SSL cert, it breaks the android app. "Oops: A server error occurred while trying to sign in. Please try again". I have many other apps using the same SSL setup (mkcert, local debian server) and they all work fine.

No other messages are provided. I have tried running with:

rails s -p 6000 -b 0.0.0.0
rails s -p 6000 -b 0.0.0.0 -e production

And my NGINX config, looks like:

```conf
upstream notes {
  server 127.0.0.1:6000;
}

server {
  listen       443 ssl http2;
  server_name  notes.mydomain.com;

  ssl_certificate /etc/nginx/certs/notes.mydomain.com.pem;
  ssl_certificate_key /etc/nginx/certs/notes.mydomain.com-key.pem;

  ssl_session_timeout 1d;
  ssl_session_cache shared:SSL:50m;
  ssl_session_tickets off;

  ssl_protocols TLSv1.2;
  ssl_ciphers EECDH+AESGCM:EECDH+AES;
  ssl_ecdh_curve secp384r1;
  ssl_prefer_server_ciphers on;

  ssl_stapling on;
  ssl_stapling_verify on;

  add_header Strict-Transport-Security "max-age=15768000; includeSubdomains; preload";#  
  add_header X-Frame-Options DENY;
  add_header X-Content-Type-Options nosniff;

  location / {
    expires -1;
    add_header Last-Modified "";
    if_modified_since off;
    etag off;
    proxy_pass http://notes;
    proxy_redirect off;
    proxy_buffering off;
    proxy_set_header        Host            $host;
    proxy_set_header        X-Real-IP       $remote_addr;
    proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
  }
}
```

I have also tried with various stripped down NGINX configs but still, nothing seems to work...

NOTE: The macOS app works fine with both SSL and non-SSL but the android app baulks when SSL is introduced.

**Comment by mobitar** [comment]
Is this a self signed cert? I'm not sure if it would work with that without custom configuration which you'd have to dig around for. I believe someone had this issue before, but it's related to system configuration rather than SN specific configuration.

**Comment by GastroGeek** [comment]
Yes, self-signed via [mkcert](https://github.com/FiloSottile/mkcert). I've set up all of the root certs across all devices/browsers and it all works fine with several other apps/servers running via reverse-proxy... but for some reason, this one simply blocks me out from the android app.

I guess It could still be SSL-related (in the self-signed sense)... I will try and dig _deeper_.

**Comment by seonwoolee** [comment]
Did you ever figure this out? I'm having the same issue. I have a self signed cert and I've installed the root certificate to my Android device. That root cert works properly for several other self hosted services' apps. I'm also able to sign into my self hosted Standard Notes sync server over SSL via the desktop Linux and iOS apps. 

If the Android app would provide any logs at all, it would be helpful. Right now I have literally no clue what the problem is. I checked logs on the server side and I don't see anything out of the ordinary. I tried a logcat over adb, but there's nothing useful concerning Standard Notes there either.
